File: android-host/app/src/main/java/com/huddle/host/runtime/EventBus.kt
package com.huddle.host.runtime

/**
 * Thin outbound bus: Room pushes wire JSON strings; HostServer delivers them.
 */
fun interface EventBus {
    suspend fun emit(message: OutboundMessage)
}

sealed class OutboundMessage {
    data class Broadcast(val json: String) : OutboundMessage()
    data class ToPlayer(val playerId: String, val json: String) : OutboundMessage()
}

File: harmony-host/shared/runtime/EventBus.ts
import type { BusSink, OutboundMessage } from "./types";

export class EventBus {
  private sink: BusSink;
  constructor(sink: BusSink) {
    this.sink = sink;
  }
  broadcast(json: string): void {
    this.sink({ kind: "broadcast", json });
  }
  toPlayer(playerId: string, json: string): void {
    this.sink({ kind: "toPlayer", playerId, json });
  }
}
